src/main/java/gyro/google/compute/SecurityPolicyRule.java

```diff
@@ -0,0 +1,191 @@
+        policyRule.setAction(getAction());
+        policyRule.setDescription(getDescription());
+        policyRule.setPriority(getPriority());
+        policyRule.setPreview(getPreview());
+        policyRule.setMatch(getMatch().toSecurityPolicyRuleMatcher());
+
+        return policyRule;
+    }
+
+    @Override
+    public void copyFrom(com.google.api.services.compute.model.SecurityPolicyRule securityPolicyRule) {
+        setPriority(securityPolicyRule.getPriority());
+        setDescription(securityPolicyRule.getDescription());
+        setAction(securityPolicyRule.getAction());
+        setPreview(securityPolicyRule.getPreview());
+
+        SecurityPolicyRuleMatcher matcher = newSubresource(SecurityPolicyRuleMatcher.class);
+        matcher.copyFrom(securityPolicyRule.getMatch());
+        setMatch(matcher);
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        Compute client = createComputeClient();
+
+        SecurityPolicyResource securityPolicyResource = (SecurityPolicyResource) this.parentResource();
+        Operation operation = client.securityPolicies()
+            .addRule(getProjectId(), securityPolicyResource.getName(), toSecurityPolicyRule())
+            .execute();
+        waitForCompletion(client, operation);
+    }
+
+    @Override
+    public void doUpdate(
+        GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        Compute client = createComputeClient();
+
+        com.google.api.services.compute.model.SecurityPolicyRule rule = new com.google.api.services.compute.model.SecurityPolicyRule();
+
+        if (changedFieldNames.contains("description")) {
+            rule.setDescription(getDescription());
+        }
+
+        if (changedFieldNames.contains("action")) {
+            rule.setAction(getAction());
+        }
+
+        if (changedFieldNames.contains("preview")) {
+            rule.setPreview(getPreview());
+        }
+
+        if (changedFieldNames.contains("match")) {
+            rule.setMatch(getMatch().toSecurityPolicyRuleMatcher());
+        }
+
+        SecurityPolicyResource securityPolicyResource = (SecurityPolicyResource) this.parentResource();
+        Operation operation = client.securityPolicies()
+            .patchRule(getProjectId(), securityPolicyResource.getName(), rule).set("priority", getPriority())
+            .execute();
+        waitForCompletion(client, operation);
+
+        refresh();
+    }
+
+    @Override
+    public void doDelete(GyroUI ui, State state) throws Exception {
+        Compute client = createComputeClient();
+
+        SecurityPolicyResource securityPolicyResource = (SecurityPolicyResource) this.parentResource();
+        if (getPriority() != 2147483647) {
+            Compute.SecurityPolicies.RemoveRule removeOperation = client.securityPolicies()
```

Comment:
If you import `Compute.SecurityPolicies.RemoveRule`
the this becomes
```suggestion
            RemoveRule removeOperation = client.securityPolicies()
```